### ISSUE TITLE
`make rpm` fails with `$'\r': command not found`.

### ISSUE THREAD

**Issue by googollee** [ocristopfer/cockpit-sensors-old#49]
# Problem

`make rpm` fails with the error `$'\r': command not found`.

# Reproduce

```bash
$ git clone https://github.com/ocristopfer/cockpit-sensors
Cloning into 'cockpit-sensors'...
remote: Enumerating objects: 342, done.
remote: Counting objects: 100% (147/147), done.
remote: Compressing objects: 100% (92/92), done.
remote: Total 342 (delta 47), reused 116 (delta 46), pack-reused 195
Receiving objects: 100% (342/342), 580.86 KiB | 1.42 MiB/s, done.
Resolving deltas: 100% (147/147), done.
$ cd cockpit-sensors
$ git co v1.4.3
Note: switching to 'v1.4.3'.
$ docker run --rm -it -v `pwd`:/data fedora /bin/bash
[root@922142a66ef1 /]# dnf install rpm-build make git nodejs-devel gettext libappstream-glib
....
Complete!
[root@922142a66ef1 /]# cd /data
[root@922142a66ef1 data]# git config --global --add safe.directory /data
[root@922142a66ef1 data]# make rpm
.... // npm outputs
git archive '54f2fd58a3645c4a222e2b1b9b5f1b9321bed998^{tree}' -- pkg/lib test/common tools/git-utils.sh tools/make-bots  | tar x
NODE_ENV=production node_modules/.bin/webpack
webpack compiled successfully

sed -e 's/%{VERSION}/1/g' packaging/cockpit-sensors.spec.in > cockpit-sensors.spec
if type appstream-util >/dev/null 2>&1; then appstream-util validate-relax --nonet *.metainfo.xml; fi
org.cockpit-project.sensors.metainfo.xml: OK
tar --xz --sort=name --mtime "@1689363724" --mode=go=rX,u+rw,a-s --numeric-owner --owner=0 --group=0 -cf cockpit-sensors-1.tar.xz --transform 's,^,cockpit-sensors/,' \
        --exclude packaging/cockpit-sensors.spec.in --exclude node_modules \
        $(git ls-files) pkg/lib test/common tools/git-utils.sh tools/make-bots  package-lock.json cockpit-sensors.spec dist/
tar --xz --sort=name --mtime "@1689363724" --mode=go=rX,u+rw,a-s --numeric-owner --owner=0 --group=0 -cf cockpit-sensors-node-1.tar.xz node_modules
mkdir -p "`pwd`/output"
mkdir -p "`pwd`/rpmbuild"
rpmbuild -bb \
  --define "_sourcedir `pwd`" \
  --define "_specdir `pwd`" \
  --define "_builddir `pwd`/rpmbuild" \
  --define "_srcrpmdir `pwd`" \
  --define "_rpmdir `pwd`/output" \
  --define "_buildrootdir `pwd`/build" \
  cockpit-sensors.spec
warning: source_date_epoch_from_changelog set but %changelog is missing
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.JbaWK5
+ umask 022
+ cd /data/rpmbuild
+ cd /data/rpmbuild
+ rm -rf cockpit-sensors
+ /usr/lib/rpm/rpmuncompress -x /data/cockpit-sensors-1.tar.xz
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd cockpit-sensors
+ rm -rf /data/rpmbuild/cockpit-sensors-SPECPARTS
+ /usr/bin/mkdir -p /data/rpmbuild/cockpit-sensors-SPECPARTS
+ /usr/lib/rpm/rpmuncompress -x /data/cockpit-sensors-node-1.tar.xz
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ rm -rf $'dist\r'
+ $'\r'
/var/tmp/rpm-tmp.JbaWK5: line 57: $'\r': command not found
error: Bad exit status from /var/tmp/rpm-tmp.JbaWK5 (%prep)

RPM build warnings:
    source_date_epoch_from_changelog set but %changelog is missing

RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.JbaWK5 (%prep)
make: *** [Makefile:145: rpm] Error 1
```

**Comment by ocristopfer** [comment]
you have try run make before mke rpm ?

**Comment by googollee** [comment]
It's same.

```bash
// prepare env and clone v1.4.3

[root@3b64c5a16f27 data]# ls
LICENSE  Makefile  README.md  cockpit-sensors.deb  org.cockpit-project.sensors.metainfo.xml  package.json  packaging  packit.yaml  plans  po  src  test  webpack.config.js
[root@3b64c5a16f27 data]# make
# if it exists already, npm install won't update it; force that so that we always get up-to-date packages
rm -f package-lock.json
# unset NODE_ENV, skips devDependencies otherwise
env -u NODE_ENV npm install --ignore-scripts
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
npm WARN deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
npm WARN deprecated @babel/plugin-proposal-private-methods@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-private-methods instead.
npm WARN deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
npm WARN deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

added 816 packages, and audited 817 packages in 1m

169 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice 
npm notice New minor version of npm available! 10.2.3 -> 10.3.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.3.0
npm notice Run npm install -g npm@10.3.0 to update!
npm notice 
env -u NODE_ENV npm prune

up to date, audited 817 packages in 2s

169 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
remote: Enumerating objects: 1472, done.
remote: Counting objects: 100% (1472/1472), done.
remote: Compressing objects: 100% (1295/1295), done.
remote: Total 1472 (delta 258), reused 507 (delta 90), pack-reused 0
Receiving objects: 100% (1472/1472), 7.22 MiB | 6.76 MiB/s, done.
Resolving deltas: 100% (258/258), done.
git archive '54f2fd58a3645c4a222e2b1b9b5f1b9321bed998^{tree}' -- pkg/lib test/common tools/git-utils.sh tools/make-bots  | tar x
NODE_ENV= node_modules/.bin/webpack
webpack compiled successfully
One of your dependencies, babel-preset-react-app, is importing the
"@babel/plugin-proposal-private-property-in-object" package without
declaring it in its dependencies. This is currently working because
"@babel/plugin-proposal-private-property-in-object" is already in your
node_modules folder for unrelated reasons, but it may break at any time.

babel-preset-react-app is part of the create-react-app project, which
is not maintianed anymore. It is thus unlikely that this bug will
ever be fixed. Add "@babel/plugin-proposal-private-property-in-object" to
your devDependencies to work around this error. This will make this message
go away.
  
[root@3b64c5a16f27 data]# make rpm
sed -e 's/%{VERSION}/1/g' packaging/cockpit-sensors.spec.in > cockpit-sensors.spec
if type appstream-util >/dev/null 2>&1; then appstream-util validate-relax --nonet *.metainfo.xml; fi
org.cockpit-project.sensors.metainfo.xml: OK
tar --xz --sort=name --mtime "@1689363724" --mode=go=rX,u+rw,a-s --numeric-owner --owner=0 --group=0 -cf cockpit-sensors-1.tar.xz --transform 's,^,cockpit-sensors/,' \
	--exclude packaging/cockpit-sensors.spec.in --exclude node_modules \
	$(git ls-files) pkg/lib test/common tools/git-utils.sh tools/make-bots  package-lock.json cockpit-sensors.spec dist/
tar --xz --sort=name --mtime "@1689363724" --mode=go=rX,u+rw,a-s --numeric-owner --owner=0 --group=0 -cf cockpit-sensors-node-1.tar.xz node_modules
mkdir -p "`pwd`/output"
mkdir -p "`pwd`/rpmbuild"
rpmbuild -bb \
  --define "_sourcedir `pwd`" \
  --define "_specdir `pwd`" \
  --define "_builddir `pwd`/rpmbuild" \
  --define "_srcrpmdir `pwd`" \
  --define "_rpmdir `pwd`/output" \
  --define "_buildrootdir `pwd`/build" \
  cockpit-sensors.spec
warning: source_date_epoch_from_changelog set but %changelog is missing
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.ABg8rc
+ umask 022
+ cd /data/rpmbuild
+ cd /data/rpmbuild
+ rm -rf cockpit-sensors
+ /usr/lib/rpm/rpmuncompress -x /data/cockpit-sensors-1.tar.xz
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd cockpit-sensors
+ rm -rf /data/rpmbuild/cockpit-sensors-SPECPARTS
+ /usr/bin/mkdir -p /data/rpmbuild/cockpit-sensors-SPECPARTS
+ /usr/lib/rpm/rpmuncompress -x /data/cockpit-sensors-node-1.tar.xz
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ rm -rf $'dist\r'
+ $'\r'
/var/tmp/rpm-tmp.ABg8rc: line 57: $'\r': command not found
error: Bad exit status from /var/tmp/rpm-tmp.ABg8rc (%prep)

RPM build warnings:
    source_date_epoch_from_changelog set but %changelog is missing

RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.ABg8rc (%prep)
make: *** [Makefile:145: rpm] Error 1
```

**Comment by googollee** [comment]
Just pr the fix.
